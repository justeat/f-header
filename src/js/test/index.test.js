import TestUtils from 'js-test-buddy';
import { setupHeader } from '../';


describe('module', () => {

    it('is a function', () => {
        expect(typeof setupHeader).toBe('function');
    });

});

describe('setupHeader', () => {

    it('converts nav checkbox to button with correct type attribute', () => {
        // Arrange
        TestUtils.setBodyHtml(`
            <input data-nav-enhance />
        `);

        // Act
        setupHeader();

        // Assert
        const html = TestUtils.getBodyHtml();
        expect(html).toMatchSnapshot();
    });

    it('adds `is-visible` class to nav container', () => {
        // Arrange
        TestUtils.setBodyHtml(`
            <input data-nav-enhance />
            <div data-nav-container></div>
        `);
        setupHeader();
        const button = document.querySelector('button[data-nav-enhance]');

        // Act
        TestUtils.click(button);

        // Assert
        const html = TestUtils.getBodyHtml();
        expect(html).toMatchSnapshot();
    });

    it('adds `is-open` class to nav label', () => {
        // Arrange
        TestUtils.setBodyHtml(`
            <input data-nav-enhance />
            <label data-nav-toggle>Menu</label>
        `);
        setupHeader();
        const button = document.querySelector('button[data-nav-enhance]');

        // Act
        TestUtils.click(button);

        // Assert
        const html = TestUtils.getBodyHtml();
        expect(html).toMatchSnapshot();
    });

    it('does nothing if nav input does not exist', () => {
        // Arrange
        TestUtils.setBodyHtml(`
            <button></button>
            <div data-nav-container></div>
            <label data-nav-toggle>Menu</label>
        `);
        setupHeader();
        const button = document.querySelector('button');

        // Act
        TestUtils.click(button);

        // Assert
        const html = TestUtils.getBodyHtml();
        expect(html).toMatchSnapshot();
    });

    it('adds `is-navInView` class to html element', () => {
        // Arrange
        TestUtils.setBodyHtml(`
            <input data-nav-enhance />
        `);
        setupHeader();
        const button = document.querySelector('button[data-nav-enhance]');

        // Act
        TestUtils.click(button);

        // Assert
        const html = document.documentElement.outerHTML;
        expect(html).toMatchSnapshot();
    });
});
