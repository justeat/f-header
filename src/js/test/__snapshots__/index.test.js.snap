// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setupHeader adds \`is-navInView\` class to html element 1`] = `
"<html class=\\"is-navInView\\"><head></head><body>
            <button data-nav-enhance=\\"\\" type=\\"button\\"></button>
        </body></html>"
`;

exports[`setupHeader adds \`is-open\` class to nav label 1`] = `
"
            <button data-nav-enhance=\\"\\" type=\\"button\\"></button>
            <label data-nav-toggle=\\"\\" class=\\"is-open\\">Menu</label>
        "
`;

exports[`setupHeader adds \`is-visible\` class to nav container 1`] = `
"
            <button data-nav-enhance=\\"\\" type=\\"button\\"></button>
            <div data-nav-container=\\"\\" class=\\"is-visible\\"></div>
        "
`;

exports[`setupHeader converts nav checkbox to button with correct type attribute 1`] = `
"
            <button data-nav-enhance=\\"\\" type=\\"button\\"></button>
        "
`;

exports[`setupHeader does nothing if nav input does not exist 1`] = `
"
            <button></button>
            <div data-nav-container=\\"\\"></div>
            <label data-nav-toggle=\\"\\">Menu</label>
        "
`;
